omen2/relation.py

```diff
@@ -0,0 +1,91 @@
+from typing import TypeVar, Callable, Iterable, TYPE_CHECKING, List
+
+from .selectable import Selectable
+
+if TYPE_CHECKING:
+    from omen2 import ObjBase, Omen, Table
+    from typing import Type
+
+T = TypeVar("T")
+
+
+# noinspection PyProtectedMember,PyDefaultArgument
+class Relation(Selectable[T]):
+    # pylint: disable=protected-access, dangerous-default-value
+
+    table_type: "Type[Table[T]]" = None
+
+    @property
+    def row_type(self):
+        return self.table_type.row_type
+
+    def __init__(self, _from: "ObjBase", _init=None, *, where=None, cascade):
+        self.cascade = cascade
+
+        self._from = _from
+        self._where = where
+        self.__table = None
+        self.__saved: List["ObjBase"] = []
+        if _init:
+            for ent in _init:
+                if isinstance(ent, dict):
+                    self.add(T(ent))
+                else:
+                    self.add(ent)
+
+    def is_bound(self):
+        return self._from._meta and self._from._meta.table
+
+    @property
+    def table(self):
+        if not self.__table:
+            mgr: "Omen" = self._from._meta.table.manager
+            self.__table: "Table" = mgr.get_table_by_name(self.table_type.table_name)
+            self.table_type = type(self.__table)
+        return self.__table
+
+    def add(self, obj: "ObjBase"):
+        if not self.is_bound():
+            self.__saved.append(obj)
+        else:
+            self._link_obj(obj)
+            self.table.add(obj)
+
+    def remove(self, obj: "ObjBase"):
+        self.table.remove(obj)
+
+    def __len__(self):
+        return sum(1 for _ in self.select())
+
+    def select(self, _where={}, **kws) -> Iterable[T]:
+        where = {**_where, **kws, **self._where}
+        for k, v in where.items():
+            if isinstance(v, Callable):
```

Comment:
am I missing somewhere we're passing callable values? Don't quite understand why you'd want that

Reply:
this happens when you want to refer to a object-property that is not a fixed-value, normally in many-to-many relationships, the where clause might be:   `{"car_id": lambda: self.id}`

Reply:
from a test:
```
self.drivers = gen_objs.drivers_relation(
            self, where={"id": lambda: self.driverid}, cascade=False
        )
```

Reply:
(this also allows relations to be based on calculated data)